e2e/specs/start-and-end-visit.spec.ts

```diff
@@ -101,6 +106,10 @@ test('Start and end a new visit', async ({ page, patient, api }) => {
   });
 
   await test.step('Then I should see a success notification', async () => {
+    // Wait for the visit end API response to complete
+    await chartPage.page.waitForResponse(response => 
+      response.url().includes('/visit') && (response.status() === 200 || response.status() === 204)
+    );
```

Comment:
The explicit waits are non-canonical here if the tests aren't timing out non-deterministically. The selector fix should solve the flakiness. If not, then we could consider adding the checks.